be/src/vec/olap/block_reader.cpp

```diff
@@ -283,8 +283,7 @@ Status BlockReader::_unique_key_next_block(Block* block, MemPool* mem_pool, Obje
 
     // do filter detete row in base compaction, only base compaction need to do the job
     if (_filter_delete) {
-        int delete_sign_idx =
-                (_sequence_col_idx == -1) ? target_columns.size() - 1 : target_columns.size() - 2;
```

Comment:
should also modify non-vectorized engine's code.

Reply:
Non-vectorized engine don't need to modify. They have different implementations that non-vectorized engine don't use delete_sign_idx.